test/integration/integration.cc

```diff
@@ -356,7 +354,11 @@ void BaseIntegrationTest::createTestServer(const std::string& json_path,
   registerTestServerPorts(port_names);
 }
 
-void BaseIntegrationTest::sendRequestAndWaitForResponse(Http::TestHeaderMapImpl& request_headers,
+HttpIntegrationTest::HttpIntegrationTest(Http::CodecClient::Type client_protocol,
```

Comment:
Do you want to split/rename to http_integation_test.cc etc?

Reply:
Sure.  I figured diffs would be easier this way, but either way it'll be separate commits.  Will do in the next push :-)

Reply:
Split out but I'll admit I was lazy cleaning up includes and libraries, given it's test code.
Feel free to tell me to be non-lazy if you care about such things, otherwise I'll just make sure the OSX build doesn't choke.

Reply:
My preference is to try to keep the test code cleanly as we do with the production code, if you don't mind doing those cleanups. If you are too busy I would potentially just add a TODO for the cleanup.